services/remotecluster/error.go

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package remotecluster
+
+import "fmt"
+
+type BufferFullError struct {
```

Comment:
This is a personal pref but i prefer using `private struct + interface` combo when the struct is not accessed directly from outside of the package boundaries. Your call, 1/5 here, just throwing it on the table.

Something like:

```go
type BufferFullError interface {
	error
	Capacity() int
}

type bufferFullError struct {
	capacity int
}
```

Reply:
What's the value adding an interface here?  If someone wants to check for this specific error type outside the package, idiomatic Go suggests they create their own interface with `Capacity()` and type assert against that.  What am I missing?

Reply:
Totally, as I said this is more a personal pref, the main benefit that I've seen is testability + extensibility like I can generate a mock from the BufferFullError interface that I can use everywhere that I need a buffer error. Also, others that need to extend the BufferFullError for example thy can just include the interface in their own interface definition and extend/add any method they need.

As I said earlier, doing this for this implementation here, is probably an overkill.